File: FrontEnd/src/data/MaliciousActivity.ts
import type { User } from './User'
import type { ActivityType } from './UserActivity'

export type MaliciousActivity = {
  id: number
  user: User | null
  activityType: ActivityType | null
  userId: number
  activityTypeId: number
  ipAddress: string
  info: string
}

export function createMaliciousActivity(data: Partial<MaliciousActivity>): MaliciousActivity {
  return {
    id: data.id ?? 0,
    user: null,
    activityType: null,
    userId: data.userId ?? 0,
    activityTypeId: data.activityTypeId ?? 0,
    ipAddress: 'no',
    info: data.info ?? '',
  }
}

File: BackEnd/PolyplayAPI/Models/Logging/MaliciousActivity.cs
using PolyplayAPI.Models.Auth;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PolyplayAPI.Models.Logging
{
    public class MaliciousActivity
    {
        [Key]
        public long Id { get; set; }
        public string UserName { get; set; }
        [ForeignKey(nameof(ActivityType))]
        public long ActivityTypeId { get; set; }

        public ActivityType ActivityType { get; set; } = null!;
        public User User { get; set; } = null!;
        public string? IpAddress { get; set; }
        public string? Info { get; set; }
    }
}
